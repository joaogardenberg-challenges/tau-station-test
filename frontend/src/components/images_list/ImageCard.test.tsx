import { render } from '@testing-library/react'
import Providers from 'providers'

const useSelector = jest.fn()
const useOnScreen = jest.fn()

jest.doMock('react-redux', () => ({ useSelector }))
jest.doMock('hooks/useOnScreen', () => useOnScreen)

const ImageCard = require('./ImageCard').default

const renderWithProviders = (ui: any, options: any = {}) =>
  render(ui, { wrapper: Providers, ...options })

describe('Image Filler', () => {
  const props = {
    imageId: 1,
    perRow: 5,
    perColumn: 3
  }

  beforeEach(() => {
    useSelector.mockReturnValue({
      id: props.imageId,
      urls: { 400: 'url' },
      meta: { keywords: 'keywords' }
    })

    useOnScreen.mockReturnValue(true)
  })

  it('renders successfully', () => {
    const { container } = renderWithProviders(<ImageCard {...props} />)
    expect(container).toMatchSnapshot()
  })
})
