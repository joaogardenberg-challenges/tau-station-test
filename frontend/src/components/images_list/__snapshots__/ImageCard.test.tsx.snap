// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image Filler renders successfully 1`] = `
<div>
  <li
    class="sc-bdfBQB bgnNMx transition-appear transition-appear-active"
  >
    <div
      class="content"
    >
      <img
        alt="keywords"
        class="loading"
        src="url"
      />
      <span
        class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-lepzwj-MuiCircularProgress-root"
        role="progressbar"
        style="width: 35px; height: 35px;"
      >
        <svg
          class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
            cx="44"
            cy="44"
            fill="none"
            r="20"
            stroke-width="4"
          />
        </svg>
      </span>
      <div
        class="overlay"
      >
        <a
          class="MuiButton-root MuiButton-text MuiButton-textSecondary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root css-11zs9w5-MuiButtonBase-root-MuiButton-root"
          href="/images/1?width=48"
          tabindex="0"
        >
          48
          px
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </a>
        <a
          class="MuiButton-root MuiButton-text MuiButton-textSecondary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root css-11zs9w5-MuiButtonBase-root-MuiButton-root"
          href="/images/1?width=400"
          tabindex="0"
        >
          400
          px
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </a>
        <a
          class="MuiButton-root MuiButton-text MuiButton-textSecondary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root css-11zs9w5-MuiButtonBase-root-MuiButton-root"
          href="/images/1?width=800"
          tabindex="0"
        >
          800
          px
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </a>
        <a
          class="MuiButton-root MuiButton-text MuiButton-textSecondary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root css-11zs9w5-MuiButtonBase-root-MuiButton-root"
          href="/images/1?width=1280"
          tabindex="0"
        >
          1280
          px
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </a>
      </div>
    </div>
  </li>
</div>
`;
